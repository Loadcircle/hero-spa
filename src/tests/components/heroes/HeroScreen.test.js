import { mount } from "enzyme";
import React from 'react';
import { MemoryRouter, Route } from "react-router-dom";
import { HeroScreen } from "../../../components/heroes/HeroScreen";

describe('Pruebas en Hero Screen', ()=>{

    const historyMock = {
        length: 10,
        push: jest.fn(),
        goBack: jest.fn(),
    }


    test('Debe de mostrar el comoponente redirect si no hay argumentos en el URL', ()=>{

        const wrapper = mount(
            //Initial entries envia parametros
            <MemoryRouter initialEntries={['/hero']}>
                <HeroScreen history={historyMock} />
            </MemoryRouter>
        );
        expect( wrapper).toMatchSnapshot();
        expect(wrapper.find('Redirect').exists()).toBe(true)

    });

    test('debe de mostrar un hero si el parametro existe', ()=>{

        const wrapper = mount(
            //Initial entries envia parametros
            <MemoryRouter initialEntries={['/hero/dc-batman']}>
                <Route path="/hero/:heroeId" component={HeroScreen} />                
            </MemoryRouter>
        );

        expect( wrapper).toMatchSnapshot();
        expect(wrapper.find('.row').exists()).toBe(true)

    });

    test('Debe regresar a la pantalla anterior con push', ()=>{

        const historyMock = {
            length: 0,
            push: jest.fn(),
            goBack: jest.fn(),
        }

        const wrapper = mount(
            //Initial entries envia parametros
            <MemoryRouter initialEntries={['/hero/dc-batman']}>
                <Route 
                    path="/hero/:heroeId" 
                    component={()=> <HeroScreen history={historyMock} />} 
                />                
            </MemoryRouter>
        );

        wrapper.find('button').prop('onClick')();

        expect(historyMock.push).toHaveBeenCalledWith('/');
        expect(historyMock.goBack).not.toHaveBeenCalled();

    });

    test('Debe regresar a la pantalla anterior con goBack', ()=>{

        const historyMock = {
            length: 5,
            push: jest.fn(),
            goBack: jest.fn(),
        }

        const wrapper = mount(
            //Initial entries envia parametros
            <MemoryRouter initialEntries={['/hero/dc-batman']}>
                <Route 
                    path="/hero/:heroeId" 
                    component={()=> <HeroScreen history={historyMock} />} 
                />                
            </MemoryRouter>
        );

        wrapper.find('button').prop('onClick')();

        expect(historyMock.goBack).toHaveBeenCalled();
        expect(historyMock.push).not.toHaveBeenCalled();
        
    });

});