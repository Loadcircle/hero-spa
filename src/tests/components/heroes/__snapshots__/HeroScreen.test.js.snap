// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en Hero Screen Debe de mostrar el comoponente redirect si no hay argumentos en el URL 1`] = `""`;

exports[`Pruebas en Hero Screen debe de mostrar un hero si el parametro existe 1`] = `
<div
  className="row mt-5  animate__animated animate__fadeIn"
>
  <div
    className="col-4"
  >
    <img
      alt="Batman"
      className="img-thumbnail"
      src="../assets/heroes/dc-batman.jpg"
    />
  </div>
  <div
    className="col-8"
  >
    <h3>
      Batman
    </h3>
    <ul
      className="list-group list-group-flush"
    >
      <li
        className="list-group-item"
      >
        <b>
          Alter ego 
        </b>
         
        Bruce Wayne
      </li>
      <li
        className="list-group-item"
      >
        <b>
          Publisher 
        </b>
         
        DC Comics
      </li>
      <li
        className="list-group-item"
      >
        <b>
          First appearance 
        </b>
         
        Detective Comics #27
      </li>
    </ul>
    <h5>
       Characters 
    </h5>
    <p>
      Bruce Wayne
    </p>
    <button
      className="btn btn-outline-primary"
      onClick={[Function]}
    >
      Return
    </button>
  </div>
</div>
`;
